### ISSUE TITLE
Crash in diagnostics analyzer

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#8178]
Was editing in roslyn when i got this:

```
Framework Version: v4.0.30319
Description: The application requested process termination through System.Environment.FailFast(string message).
Message: System.InvalidOperationException: Cannot call Enqueue when the queue is already completed.
   at Microsoft.CodeAnalysis.Diagnostics.AsyncQueue`1.Enqueue(TElement value)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.SymbolDeclaredEvent(Symbol symbol)
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceNamespaceSymbol.GetNameToMembersMap()
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceNamespaceSymbol.GetMembers(String name)
   at Microsoft.CodeAnalysis.CSharp.SyntaxTreeSemanticModel.GetDeclaredMember(NamespaceOrTypeSymbol container, TextSpan declarationSpan, String name)
   at Microsoft.CodeAnalysis.CSharp.SyntaxTreeSemanticModel.GetDeclaredNamedType(CSharpSyntaxNode declarationSyntax, String name)
   at Microsoft.CodeAnalysis.CSharp.SyntaxTreeSemanticModel.GetDeclaredType(BaseTypeDeclarationSyntax declarationSyntax)
   at Microsoft.CodeAnalysis.CSharp.SyntaxTreeSemanticModel.GetDeclaredSymbol(BaseTypeDeclarationSyntax declarationSyntax, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel.GetDeclaredSymbolCore(SyntaxNode declaration, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.DeclarationComputer.GetDeclaredSymbol(SemanticModel model, SyntaxNode node, Boolean getSymbol, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.DeclarationComputer.GetDeclarationInfo(SemanticModel model, SyntaxNode node, Boolean getSymbol, IEnumerable`1 executableCodeBlocks, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.DeclarationComputer.GetDeclarationInfo(SemanticModel model, SyntaxNode node, Boolean getSymbol, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpDeclarationComputer.ComputeDeclarations(SemanticModel model, SyntaxNode node, Func`3 shouldSkip, Boolean getSymbol, List`1 builder, Nullable`1 levelsToCompute, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpDeclarationComputer.ComputeDeclarations(SemanticModel model, SyntaxNode node, Func`3 shouldSkip, Boolean getSymbol, List`1 builder, Nullable`1 levelsToCompute, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpDeclarationComputer.ComputeDeclarationsInNode(SemanticModel model, SyntaxNode node, Boolean getSymbol, List`1 builder, CancellationToken cancellationToken, Nullable`1 levelsToCompute)
   at Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel.ComputeDeclarationsInNode(SyntaxNode node, Boolean getSymbol, List`1 builder, CancellationToken cancellationToken, Nullable`1 levelsToCompute)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.ComputeDeclarationAnalysisData(ISymbol symbol, SyntaxReference declaration, SemanticModel semanticModel, Boolean shouldExecuteSyntaxNodeActions, AnalysisScope analysisScope, Func`1 allocateData, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.<>c__DisplayClass37_0.<ExecuteDeclaringReferenceActions>b__0(Func`1 allocateData)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.CompilationData.GetOrComputeDeclarationAnalysisData(SyntaxReference declaration, Func`2 computeDeclarationAnalysisData, Boolean cacheAnalysisData)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.ExecuteDeclaringReferenceActions(SyntaxReference decl, Int32 declarationIndex, SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, Boolean shouldExecuteSyntaxNodeActions, Boolean shouldExecuteOperationActions, Boolean shouldExecuteCodeBlockActions, Boolean shouldExecuteOperationBlockActions, Boolean isInGeneratedCode, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.ExecuteDeclaringReferenceActions(SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, Boolean isGeneratedCodeSymbol, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessSymbolDeclared(SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessEventCore(CompilationEvent e, AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.<ProcessCompilationEventsCoreAsync>d__64.MoveNext()
Stack:
   at System.Environment.FailFast(System.String, System.Exception)
   at Microsoft.CodeAnalysis.FatalError.Report(System.Exception, System.Action`1<System.Exception>)
   at Microsoft.CodeAnalysis.FatalError.ReportUnlessCanceled(System.Exception)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<ProcessCompilationEventsCoreAsync>d__64.MoveNext()
   at Microsoft.CodeAnalysis.Diagnostics.AsyncQueue`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].Enqueue(System.__Canon)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.SymbolDeclaredEvent(Microsoft.CodeAnalysis.CSharp.Symbol)
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceNamespaceSymbol.GetNameToMembersMap()
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceNamespaceSymbol.GetMembers(System.String)
   at Microsoft.CodeAnalysis.CSharp.SyntaxTreeSemanticModel.GetDeclaredMember(Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceOrTypeSymbol, Microsoft.CodeAnalysis.Text.TextSpan, System.String)
   at Microsoft.CodeAnalysis.CSharp.SyntaxTreeSemanticModel.GetDeclaredNamedType(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode, System.String)
   at Microsoft.CodeAnalysis.CSharp.SyntaxTreeSemanticModel.GetDeclaredType(Microsoft.CodeAnalysis.CSharp.Syntax.BaseTypeDeclarationSyntax)
   at Microsoft.CodeAnalysis.CSharp.SyntaxTreeSemanticModel.GetDeclaredSymbol(Microsoft.CodeAnalysis.CSharp.Syntax.BaseTypeDeclarationSyntax, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel.GetDeclaredSymbolCore(Microsoft.CodeAnalysis.SyntaxNode, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.DeclarationComputer.GetDeclaredSymbol(Microsoft.CodeAnalysis.SemanticModel, Microsoft.CodeAnalysis.SyntaxNode, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.DeclarationComputer.GetDeclarationInfo(Microsoft.CodeAnalysis.SemanticModel, Microsoft.CodeAnalysis.SyntaxNode, Boolean, System.Collections.Generic.IEnumerable`1<Microsoft.CodeAnalysis.SyntaxNode>, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.DeclarationComputer.GetDeclarationInfo(Microsoft.CodeAnalysis.SemanticModel, Microsoft.CodeAnalysis.SyntaxNode, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpDeclarationComputer.ComputeDeclarations(Microsoft.CodeAnalysis.SemanticModel, Microsoft.CodeAnalysis.SyntaxNode, System.Func`3<Microsoft.CodeAnalysis.SyntaxNode,System.Nullable`1<Int32>,Boolean>, Boolean, System.Collections.Generic.List`1<Microsoft.CodeAnalysis.DeclarationInfo>, System.Nullable`1<Int32>, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpDeclarationComputer.ComputeDeclarations(Microsoft.CodeAnalysis.SemanticModel, Microsoft.CodeAnalysis.SyntaxNode, System.Func`3<Microsoft.CodeAnalysis.SyntaxNode,System.Nullable`1<Int32>,Boolean>, Boolean, System.Collections.Generic.List`1<Microsoft.CodeAnalysis.DeclarationInfo>, System.Nullable`1<Int32>, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpDeclarationComputer.ComputeDeclarationsInNode(Microsoft.CodeAnalysis.SemanticModel, Microsoft.CodeAnalysis.SyntaxNode, Boolean, System.Collections.Generic.List`1<Microsoft.CodeAnalysis.DeclarationInfo>, System.Threading.CancellationToken, System.Nullable`1<Int32>)
   at Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel.ComputeDeclarationsInNode(Microsoft.CodeAnalysis.SyntaxNode, Boolean, System.Collections.Generic.List`1<Microsoft.CodeAnalysis.DeclarationInfo>, System.Threading.CancellationToken, System.Nullable`1<Int32>)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1[[Microsoft.CodeAnalysis.CSharp.SyntaxKind, Microsoft.CodeAnalysis.CSharp, Version=1.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]].ComputeDeclarationAnalysisData(Microsoft.CodeAnalysis.ISymbol, Microsoft.CodeAnalysis.SyntaxReference, Microsoft.CodeAnalysis.SemanticModel, Boolean, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, System.Func`1<DeclarationAnalysisData>, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1+<>c__DisplayClass37_0[[Microsoft.CodeAnalysis.CSharp.SyntaxKind, Microsoft.CodeAnalysis.CSharp, Version=1.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]].<ExecuteDeclaringReferenceActions>b__0(System.Func`1<DeclarationAnalysisData>)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+CompilationData.GetOrComputeDeclarationAnalysisData(Microsoft.CodeAnalysis.SyntaxReference, System.Func`2<System.Func`1<DeclarationAnalysisData>,DeclarationAnalysisData>, Boolean)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1[[Microsoft.CodeAnalysis.CSharp.SyntaxKind, Microsoft.CodeAnalysis.CSharp, Version=1.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]].ExecuteDeclaringReferenceActions(Microsoft.CodeAnalysis.SyntaxReference, Int32, Microsoft.CodeAnalysis.Diagnostics.SymbolDeclaredCompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, Boolean, Boolean, Boolean, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1[[Microsoft.CodeAnalysis.CSharp.SyntaxKind, Microsoft.CodeAnalysis.CSharp, Version=1.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]].ExecuteDeclaringReferenceActions(Microsoft.CodeAnalysis.Diagnostics.SymbolDeclaredCompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessSymbolDeclared(Microsoft.CodeAnalysis.Diagnostics.SymbolDeclaredCompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessEventCore(Microsoft.CodeAnalysis.Diagnostics.CompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<ProcessCompilationEventsCoreAsync>d__64.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].Start[[Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<ProcessCompilationEventsCoreAsync>d__64, Microsoft.CodeAnalysis, Version=1.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]](<ProcessCompilationEventsCoreAsync>d__64 ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessCompilationEventsCoreAsync(Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<>c__DisplayClass63_0+<<ProcessCompilationEventsAsync>b__0>d.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].Start[[Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<>c__DisplayClass63_0+<<ProcessCompilationEventsAsync>b__0>d, Microsoft.CodeAnalysis, Version=1.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]](<<ProcessCompilationEventsAsync>b__0>d ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<>c__DisplayClass63_0.<ProcessCompilationEventsAsync>b__0()
   at System.Threading.Tasks.Task`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef)
   at System.Threading.Tasks.Task.ExecuteEntry(Boolean)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

**Comment by mavasani** [comment]
This is a compiler issue - we seem to be generating a symbol declared event after compilation end event. Cyrus, do you have repro?

@gafter may want to take a look.

**Comment by CyrusNajmabadi** [comment]
Nope.  I was just editing, and poof, down went VS.  Hopefully a watson got sent.

**Comment by jaredpar** [comment]
Removed the urgency label.  From what I'm reading in the bug this is a one time repro that isn't actively blocking anyone.

**Comment by gafter** [comment]
I suspect what is happening here is due to recovery in the case of a syntax error. The semantic model can't find an appropriate class for the declaration, so it makes one.

I think the appropriate recovery is to simply stop producing events after the symbol table is complete.

**Comment by gafter** [comment]
@jaredpar Assigning to you as you have the PR in ask mode.

**Comment by jmarolf** [comment]
Can you link to the PR that fixed this issue?

**Comment by gafter** [comment]
This was fixed in #8836
